test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIntegrationTest.cs

```diff
@@ -367,6 +367,51 @@ public Task Build_SimpleMvc_UsingDesktopMSBuildAndWithoutBuildServer_CanBuildSuc
             Assert.FileContains(result, razorAssemblyInfoPath, "[assembly: Microsoft.AspNetCore.Mvc.ApplicationParts.ProvideApplicationPartFactoryAttribute(\"Microsoft.AspNetCore.Mvc.ApplicationParts.CompiledRazorAssemblyApplicationPartFac\"");
         }
 
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Build_DoesNotAddRelatedAssemblyPart_IfToolSetIsNotRazorSdk()
+        {
+            var assemblyInfo = Path.Combine(IntermediateOutputPath, "SimpleMvc.AssemblyInfo.cs");
+            var result = await DotnetMSBuild("Build", "/p:RazorCompileToolSet=MvcPrecompilation");
+
+            Assert.BuildPassed(result);
+
+            Assert.FileExists(result, assemblyInfo);
+            Assert.FileDoesNotContain(result, assemblyInfo, "RelatedAssemblyAttribute");
+
+            Assert.FileDoesNotExist(result, IntermediateOutputPath, "SimpleMvc.RazorAssemblyInfo.cs");
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Build_DoesNotAddRelatedAssemblyPart_IfViewCompilationIsDisabled()
+        {
+            var assemblyInfo = Path.Combine(IntermediateOutputPath, "SimpleMvc.AssemblyInfo.cs");
+            var result = await DotnetMSBuild("Build", "/p:RazorCompileOnBuild=false /p:RazorCompileOnPublish=false");
+
+            Assert.BuildPassed(result);
+
+            Assert.FileExists(result, assemblyInfo);
+            Assert.FileDoesNotContain(result, assemblyInfo, "RelatedAssemblyAttribute");
+
+            Assert.FileDoesNotExist(result, IntermediateOutputPath, "SimpleMvc.RazorAssemblyInfo.cs");
+        }
+
+        [Fact]
+        [InitializeTestProject("SimpleMvc")]
+        public async Task Build_AddsRelatedAssemblyPart_IfCompileOnPublishIsAllowed()
```

Comment:
This scenario forces us in being defensive with loading related assembly parts i.e. no-op when the assembly cannot be found.